### PR TITLE
feat(themes): update blog theme and set up development workspace

### PR DESCRIPTION
**Pull request by jxnblk**
## Description

This sets up the `themes/` directory as a Yarn workspace for local development of themes. The `gatsby-starter-theme-blog-mdx` is meant to be a starter package for the MDX blog theme. I'm happy to change anything with regards to how this is set up, but wanted to introduce an easier way to develop the themes here.

I've also updated the `gatsby-theme-blog-mdx` package to remove `typography-system` in favor of `theme-ui` and flattened the tokens/themes since shadowing now allows the shadowed module to be imported. The styles in the theme *should* be scoped to the Layout component, meaning that the MDX and Emotion contexts are provided in `wrapRootElement` but styles are only applied when using the Layout component.

### CODE REVIEW

File: themes/gatsby-starter-theme-blog-mdx/package.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "gatsby-starter-theme-blog-mdx",
```
Comment (DSchau) [inline themes/gatsby-starter-theme-blog-mdx/package.json]: Hate to nit on naming, but should this be gatsby-theme-blog-mdx or gatsby-starter-theme-blog-mdx? The latter seems a little confusing!

Comment (jxnblk) [reply themes/gatsby-starter-theme-blog-mdx/package.json]: I really have no idea how to name this sort of thing – more than happy to change this to whatever makes most sense. To clarify, this is meant to be a starter that uses the `gatsby-theme-blog-mdx` theme

Comment (ChristopherBiscardi) [reply themes/gatsby-starter-theme-blog-mdx/package.json]: `gatsby-theme-blog-mdx`, but we're going to make the `-blog` named one MDX based anyway so the naming will change.

Comment (ChristopherBiscardi) [reply themes/gatsby-starter-theme-blog-mdx/package.json]: the two blog themes (mdx+remark) will merge into one with the data theme (this is part of what I was experimenting with on my own blog), so the starter will eventually replace https://github.com/gatsbyjs/gatsby-starter-blog. As such, it's a temporary thing in this directory and the naming probably doesn't matter much.

Comment (DSchau) [reply themes/gatsby-starter-theme-blog-mdx/package.json]: About what I expected -- thanks!

File: themes/gatsby-starter-theme-blog-mdx/src/data/author.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+- id: jay
+  name: Jay Gatsby
+  bio: A mysterious millionaire with shady business connections.
```
Comment (sidharthachatterjee) [inline themes/gatsby-starter-theme-blog-mdx/src/data/author.yml]: 😂

**Comment by sidharthachatterjee** [comment]
`starters_validate` is failing due to some (unrelated) npm audit warnings

**Pull request merged**